pkg/venafi/cloud/provisioning.go

```diff
@@ -0,0 +1,254 @@
+	CloudCertificateName       string `json:"cloudProviderCertificateName"`
+	CloudCertificateVersion    string `json:"cloudProviderCertificateVersion"`
+	Error                      error  `json:"error"`
+}
+
+type CloudProvisioningMetadata struct {
+	awsMetadata   CloudAwsMetadata
+	azureMetadata CloudAzureMetadata
+	gcpMetadata   CloudGcpMetadata
+}
+
+func (cpm *CloudProvisioningMetadata) GetAWSCertificateMetadata() endpoint.AWSCertificateMetadata {
+	return &cpm.awsMetadata
+}
+
+func (cpm *CloudProvisioningMetadata) GetAzureCertificateMetadata() endpoint.AzureCertificateMetadata {
+	return &cpm.azureMetadata
+}
+
+func (cpm *CloudProvisioningMetadata) GetGCPCertificateMetadata() endpoint.GCPCertificateMetadata {
+	return &cpm.gcpMetadata
+}
+
+type CloudAwsMetadata struct {
+	result CloudKeystoreProvisioningResult
+}
+
+func (cawm *CloudAwsMetadata) GetARN() string {
+	return cawm.result.Arn
+}
+
+type CloudGcpMetadata struct {
+	result CloudKeystoreProvisioningResult
+}
+
+func (cgm *CloudGcpMetadata) GetID() string {
+	return cgm.result.CloudProviderCertificateID
+}
+
+func (cgm *CloudGcpMetadata) GetName() string {
+	return cgm.result.CloudCertificateName
+}
+
+type CloudAzureMetadata struct {
+	result CloudKeystoreProvisioningResult
+}
+
+func (cam *CloudAzureMetadata) GetName() string {
+	return cam.result.CloudCertificateName
+}
+
+func (cam *CloudAzureMetadata) GetVersion() string {
+	return cam.result.CloudCertificateVersion
+}
+
+func (cam *CloudAzureMetadata) GetID() string {
+	return cam.result.CloudProviderCertificateID
+}
+
+// GCMCertificateScope Indicates the Scope for a certificate provisioned to GCP Certificate Manager
+type GCMCertificateScope string
+
+const (
+	// GCMCertificateScopeDefault Certificates with default scope are served from core Google data centers.
+	// If unsure, choose this option.
+	GCMCertificateScopeDefault GCMCertificateScope = "DEFAULT"
+	// GCMCertificateScopeEdgeCache Certificates with scope EDGE_CACHE are special-purposed certificates,
+	// served from Edge Points of Presence.
+	// See https://cloud.google.com/vpc/docs/edge-locations.
+	GCMCertificateScopeEdgeCache GCMCertificateScope = "EDGE_CACHE"
+)
+
+type CertificateTagOption struct {
+	Name  string
+	Value string
+}
+
+type CloudProvisioningAzureOptions struct {
+	Name       *string
+	Enabled    *bool
+	Exportable *bool
+	Reusekey   *bool
+	Tags       []*CertificateTagOption
+}
+
+func (cpao CloudProvisioningAzureOptions) GetType() string {
+	return "AKV"
+}
+
+type CloudProvisioningGCPOptions struct {
+	ID          *string
+	Description *string
+	Scope       *GCMCertificateScope
+	Labels      []*CertificateTagOption
+}
+
+func (cpgo CloudProvisioningGCPOptions) GetType() string {
+	return "GCM"
+}
+
+func (c *Connector) ProvisionCertificate(req *endpoint.ProvisioningRequest, options *endpoint.ProvisioningOptions) (provisioningMetadata endpoint.ProvisioningMetadata, err error) {
```

Comment:
I actually didn't move them in there because I really did not see them as not par of the connector, at least for:

- getGraphqlClient
- validateIfCertIsVCPGeneratedByID

Since they can be used in other parts as well of the connector

Reply:
That said, I'll do the change as you mention

Reply:
Fixed in latest commit